src/main/java/inventorysetups/InventorySetupsPersistentDataManager.java

```diff
@@ -210,26 +272,42 @@ private void handleMigrationOfOldData()
 		if (Strings.isNullOrEmpty(hasMigratedToV2))
 		{
 			log.info("Migrating data to V2");
-			// Perform migration of old data
-			Type setupType = new TypeToken<ArrayList<InventorySetup>>()
+			inventorySetups.addAll(loadV1Setups());
+			updateConfig(true, false);
+			inventorySetups.clear();
+			configManager.setConfiguration(CONFIG_GROUP, CONFIG_KEY_SETUPS_MIGRATED_V2, "True");

```

Comment:
This here is incorrect. I tested a situation where someone still had V1 data and never migrated to V2. All the setups get wiped. If a migration to V2 never happened, then we go straight to V3, and I think we can set both MIGRATED_V2 and MIGRATED_V3 to true:

```
configManager.setConfiguration(CONFIG_GROUP, CONFIG_KEY_SETUPS_MIGRATED_V2, "True");
configManager.setConfiguration(CONFIG_GROUP, CONFIG_KEY_SETUPS_MIGRATED_V3, "True");
```

Pretty niche situation but might help someone out there with V1 data who comes back to play again.

Reply:
done; great catch